### ISSUE TITLE
Changes in location case forces the resource to be replaced

### ISSUE THREAD

**Issue by ilmax** [hashicorp/terraform-provider-azurerm#19813]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.5

### AzureRM Provider Version

3.35.0

### Affected Resource(s)/Data Source(s)

Api Management, Cosmos and possibly others

### Terraform Configuration Files

```hcl
data "azurerm_resource_group" "core_resource_group" {
  name = var.core_resource_group_name
}

resource "azurerm_api_management" "api_management" {
  name = var.apim_service_name
  location            = data.azurerm_resource_group.core_resource_group.location
  ...
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

Now I don't know why this is the case and if the Azure API are starting to return inconsistent values, but I was thinking that maybe the provider can be less picky on these changes and recognize that "West Europe" and "westeurope" are effectively the same thing and treat them as such.

### Actual Behaviour

I'm migrating several resource to terraform from ARM + PowerShell, so I'm writing the modules and relative import scripts.
I've noticed that old modules (written around 1/2 months ago) started to replace resources because of a difference in the location property.

In my modules I use to read the resource group with a data source and the use that location as the resource location (See snippet below). In Azure (looking at the json) the resource group has the location set to "westeurope" while the other resources that are marked to be replaced have "West Europe".

**Comment by mpicard** [comment]
Add this to the resource

```terraform
lifecycle {
  ignore_changes = [zone]
}
```

**Comment by ilmax** [comment]
Hey @mpicard thanks for the suggestion. I'm aware of what `ignore_changes` does anyway I opened the issue to check if we instruct the azurerm provider ignore the difference rather than me opting out manually.

**Comment by sinbai** [comment]
@ilmax thanks for opening this issue. I could not repro this issue by the following tf config and repro steps. Could you provide the step-by-step-repro-steps and tf config to help repro/troubleshooting?

1. Run "terraform apply" to create a resource group by the following tf config.
```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.35.0"
    }
  }
}

provider "azurerm" {
  features {
    }
}


resource "azurerm_resource_group" "test" {
  name     = "19813"
  location = "West Europe"
}
```

2. Adding the following tf config into above config to create azurerm_api_management to read the resource group with a data source by run `terraform apply`. 
```
data "azurerm_resource_group" "core_resource_group" {
  name = "19813"
}

resource "azurerm_api_management" "api_management" {
  name = "apim0103"
  location            = data.azurerm_resource_group.core_resource_group.location
  resource_group_name ="19813"
  publisher_name      = "pub1"
  publisher_email     = "pub1@email.com"

  sku_name = "Developer_1"
}

```
3. Run `terraform plan` =>no change.
Actual:
![image](https://user-images.githubusercontent.com/39109137/210305721-145ce4a6-387c-4dd7-bd8d-853bf60eba8d.png)

**Comment by ilmax** [comment]
Hi @sinbai , as I mentioned above, I'm importing existing resources, problem is that in Azure, if I go the the resource overview and click on the JSON view link on the right it shows me different value for the location, in this specific case it shows "westeurope" for the resource group and "West Europe" for api management.

What I was asking here is if it make sense to teach azurerm terraform provider that "westeurope" and "West Europe" are effectively the same thing so there's no need to destroy and recreate the resource.

Hope this clarifies the issue

**Comment by ilmax** [comment]
@sinbai  you're right, apparently this only happened as soon as I started adding depends_on at the module level.

so I have several modules defined and I needed to enforce the creation order (not that it was actually needed since resources already exists, but I added it anyway to my HCL code) this in turn marked a lot of resources to be recreated so I had to move away from that (i.e. no more module depends_on) and now everything works.

Let me know if you're still interested into the repro steps or if I should just close the issue, for me the problem is solved at the moment
